README.rst

```diff
@@ -1,74 +1,219 @@
+
+Every time you develop something in this repo
+---------------------------------------------
+
+Activate the virtualenv. Here's how you might do that if you're using
+``virtualenv``:
+
+.. code-block:: bash
+
+  source platform_plugin_elm_credentials/bin/activate
+
+Grab the latest code:
+
+.. code-block:: bash
+
+  git checkout main
+  git pull
+
+Install/update the dev requirements:
+
+.. code-block:: bash
+
+  make requirements
+
+Run the tests and quality checks (to verify the status before you make any
+changes):
+
+.. code-block:: bash
+
+  make validate
+
+Make a new branch for your changes:
+
+.. code-block:: bash
+
+  git checkout -b <your_github_username>/<short_description>
+
+Using your favorite editor, edit the code to make your change:
+
+.. code-block:: bash
+
+  vim ...
+
+Run your new tests:
+
+.. code-block:: bash
+
+  pytest ./path/to/new/tests
+
+Run all the tests and quality checks:
+
+.. code-block:: bash
+
+  make validate
+
+Commit all your changes, push your branch to github, and open a PR:
+
+.. code-block:: bash
+
+  git commit ...
+  git push
 
-Getting Help
-************
 
-Documentation
-=============
+Using the API
+*************
 
-PLACEHOLDER: Start by going through `the documentation`_.  If you need more help see below.
+**IMPORTANT**: To use the API, you need to have a course with **certificates**
+enabled. You should consider the following:
 
-.. _the documentation: https://docs.openedx.org/projects/platform-plugin-elm-credentials
+1. You should configure from the Django administrator the **coursemode**, the
+   **certificate generation configuration**, and the **certificate html view configuration**
+2. You should activate certificates in the course from Studio in the
+   **Settings** > **Certificates**.
 
-(TODO: `Set up documentation <https://openedx.atlassian.net/wiki/spaces/DOC/pages/21627535/Publish+Documentation+on+Read+the+Docs>`_)
+Now, you can use the API. The API endpoints is protected with the same auth
+method as the Open edX platform. For generate a token, you can use the next
+endpoint:
 
-More Help
-=========
+- POST ``/<lms_host>/oauth2/access_token/``: Generate a token for the user. The
+  content type of the request must be ``application/x-www-form-urlencoded``.
 
-If you're having trouble, we have discussion forums at
-https://discuss.openedx.org where you can connect with others in the
-community.
+  **Body parameters**
 
-Our real-time conversations are on Slack. You can request a `Slack
-invitation`_, then join our `community Slack workspace`_.
+  - ``client_id``: Client ID of the OAuth2 application. You can find it in the
+    Django admin panel. Normally, it is ``login-service-client-id``.
+  - ``grant_type``: Grant type of the OAuth2 application. Normally, it is
+    ``password``.
+  - ``username``: Username of the user.
+  - ``password``: Password of the user.
+  - ``token_type``: Type of the token. By default, it is ``bearer``
```

Comment:
You should also mention that a new oauth2 application can be used

Reply:
Done